tools/database/dbc/item_effect.go

```diff
@@ -33,3 +40,193 @@ func (e *ItemEffect) ToMap() map[string]interface{} {
+	sp := dbcInstance.Spells[statsSpellID]
+	if sp.Duration > 0 {
+		pe.EffectDuration = sp.Duration / 1000
+	}
+	switch trig {
+	case ITEM_SPELLTRIGGER_ON_USE:
+		pe.Type = proto.ItemEffectType_ON_USE
+		pe.Effect = &proto.ItemEffect_OnUse{
+			OnUse: &proto.OnUseEffect{
+				Cooldown:         int32(e.CoolDownMSec / 1000),
+				CategoryId:       int32(e.SpellCategoryID),
+				CategoryCooldown: int32(e.CategoryCoolDownMSec / 1000),
+			},
+		}
+
+	case ITEM_SPELLTRIGGER_CHANCE_ON_HIT:
+		// For procchance and ICD we always use the original spell id
+		spTop := dbcInstance.Spells[e.SpellID]
+		effect := &proto.ProcEffect{
+			ProcChance: float64(spTop.ProcChance) / 100,
+			Icd:        int32(spTop.ProcCategoryRecovery / 1000),
+			Rppm:       spTop.Rppm,
+			RppmScale:  int32(realPpmScale(spTop)),
+		}
+		// On procs we want the lower name though
+		pe.Label = sp.NameLang
+		pe.SpellId = sp.ID
+		ilvlMod, specMods := realPpmModifier(spTop, itemLevel)
+		effect.SpecModifiers = specMods
+
+		pe.Type = proto.ItemEffectType_PROC
+		pe.Effect = &proto.ItemEffect_Proc{
+			Proc: effect,
+		}
+		return statsSpellID, ilvlMod
+	default:
+		// leave as NONE
+	}
+
+	return statsSpellID, 0
+}
+
+func realPpmScale(spell Spell) int {
+	scale := 0
+	for _, mod := range spell.RppmModifiers {
+		switch mod.ModifierType {
+		case RPPMModifierHaste:
+			scale |= core.RPPM_HASTE
+		case RPPMModifierCrit:
+			scale |= core.RPPM_CRIT
+		}
+	}
+	return scale
+}
+
+func realPpmModifier(spell Spell, itemLevel int) (float64, map[int32]float64) {
+	specModifier := make(map[int32]float64)
+	ilvlModifier := 1.0
+	for _, mod := range spell.RppmModifiers {
+		switch mod.ModifierType {
+		case RPPMModifierSpec:
+			spec := SpecFromID(mod.Param)
+			specModifier[int32(spec)] = 1.0 * (1.0 + mod.Coeff)
+
+		case RPPMModifierIlevel:
+			basePoints := dbcInstance.RandomPropertiesByIlvl[int(mod.Param)][proto.ItemQuality_ItemQualityRare][0]
+			ilvlPoints := dbcInstance.RandomPropertiesByIlvl[itemLevel][proto.ItemQuality_ItemQualityRare][0]
+			if basePoints != ilvlPoints {
+				ilvlModifier *= 1.0 + ((float64(ilvlPoints)/float64(basePoints))-1.0)*mod.Coeff
+			}
+		}
+	}
+	return ilvlModifier, specModifier
+}
+
+func resolveTrigger(topType, spellID int) (triggerType, statsSpellID int) {
+	if topType == ITEM_SPELLTRIGGER_ON_USE || topType == ITEM_SPELLTRIGGER_CHANCE_ON_HIT {
+		return topType, spellID
+	}
+	for _, se := range dbcInstance.SpellEffects[spellID] {
+		if se.EffectAura == A_PROC_TRIGGER_SPELL {
+			// stats come from the triggered spell
+			return resolveTrigger(ITEM_SPELLTRIGGER_CHANCE_ON_HIT, se.EffectTriggerSpell)
+		}
+	}
+	return topType, spellID
+}
+
+func buildScalingProps(spellID, itemLevel int) *proto.ScalingItemEffectProperties {
+	total := collectStats(spellID, itemLevel)
+	src := total.ToProtoMap()
+
+	m := make(map[int32]float64, len(src))
+	maps.Copy(m, src)
+
+	return &proto.ScalingItemEffectProperties{Stats: m}
```

Comment:
Can you just directly populate the `Stats` field with `src` rather than allocating another map and copying to it first?